From a7aeaa6b6073a8960b073d9b9440b4308223d6e9 Mon Sep 17 00:00:00 2001
From: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Date: Tue, 30 Apr 2013 15:16:11 +0300
Subject: [PORT FROM MAIN] Telephony: fix issue with data state reporting

BZ: 34445

In case of network lost for short duration, network may not
disconnect the data call. When the network is back, same data
connection may still be alive.

DISCONNECTED should be informed only when the network or modem informs
of the disconnection.

Change-Id: I0523dd4127430b315b3248869086956b94c5eb91
Orig-Change-Id: If71d0905be4eee8a858e02deae2ad5c8461ee9aa
Category: aosp improvement
Domain: Telephony-Telephony-FWK
Origin: internal
Upstream-Candidate: yes
Signed-off-by: Jeevaka Badrappan <jeevaka.badrappan@intel.com>
Signed-off-by: Arun Ravindran <arun.ravindran@intel.com>
---
 .../android/internal/telephony/gsm/GSMPhone.java   |    5 -----
 1 files changed, 0 insertions(+), 5 deletions(-)

diff --git a/src/java/com/android/internal/telephony/gsm/GSMPhone.java b/src/java/com/android/internal/telephony/gsm/GSMPhone.java
index 0e2a73e..9a07fd9 100644
--- a/src/java/com/android/internal/telephony/gsm/GSMPhone.java
+++ b/src/java/com/android/internal/telephony/gsm/GSMPhone.java
@@ -300,11 +300,6 @@ public class GSMPhone extends PhoneBase {
             // already been called
 
             ret = PhoneConstants.DataState.DISCONNECTED;
-        } else if (mSST.getCurrentDataConnectionState()
-                != ServiceState.STATE_IN_SERVICE) {
-            // If we're out of service, open TCP sockets may still work
-            // but no data will flow
-            ret = PhoneConstants.DataState.DISCONNECTED;
         } else if (mDcTracker.isApnTypeEnabled(apnType) == false ||
                 mDcTracker.isApnTypeActive(apnType) == false) {
             //TODO: isApnTypeActive() is just checking whether ApnContext holds
-- 
1.7.4.1

